admin/sql/CreateFunctions.sql

```diff
@@ -479,6 +479,19 @@ BEGIN
         PERFORM dec_ref_count('artist_credit', OLD.artist_credit, 1);
         PERFORM inc_ref_count('artist_credit', NEW.artist_credit, 1);
     END IF;
+    IF (
+        NEW.status IS DISTINCT FROM OLD.status AND
+        (NEW.status = 6 OR OLD.status = 6)
+    ) THEN
```

Comment:
It might be possible, it just seems like it would be less clear. I expect that's the reason we have two different checks for other things already (there's multiple `NEW.release_group != OLD.release_group` and `NEW.artist_credit != OLD.artist_credit`).

Reply:
There is only one other `NEW.status IS DISTINCT FROM OLD.status`.

Reply:
Yes, I know. My point is that this function already duplicates other checks for clarity when it could group them to save one check, so it seems sensible to keep prioritizing clarity over grouping for this new case.

Reply:
OK, it doesn’t seem to be a valid point. Grouping conditions would not only be a gain of performances, it would also be a gain of clarity as you won’t have to skim the whole function for repeated conditions. Even if it was about to make anything less clear, comments can always be added to fix that.

Reply:
IIUC, this check is `A && B` while the one below is `A || C || D`.  So combining the `A` checks means you'll have to duplicate the `INSERT INTO artist_release_group_pending_update` line.  When I was originally writing those materialized tables, I found it clearer to keep all the conditions related to updating them together, because I could quickly check that all the cases are covered in one place.  I didn't worry about performance so much because comparing two integers in PL/pgSQL should be extremely fast.

Reply:
> No INSERT duplication is needed, you can just have A && B nested under A || C || D.

I thought the point was to avoid testing A twice.

Reply:
`A && B` would not tested twice when `A || C || D` is false, which is a very (possibly the most) common case.

But this discussion already wasted more energy than it would save. Maybe one day someone will care improving either performance or clarity. Now that we are pressed by time for the schema change, this is not a showstopper for merging this pull request.